lualib/lua_scanners/icap.lua

```diff
@@ -399,37 +403,47 @@ local function icap_check(task, content, digest, rule, maybe_part)
           http: HTTP/1.0 307 Temporary Redirect
         ]] --
 
+        local function string_to_threat_table(header_string)
+          if string.find(header_string, ', ') then
+            local vnames = lua_util.str_split(string.gsub(header_string, "%s", ""), ',') or {}
+
+            for _,v in ipairs(vnames) do
+              table.insert(threat_string, v)
+            end
+          else
+            table.insert(threat_string, header_string)
+          end
+        end
+
         -- Generic ICAP Headers
         if headers['x-infection-found'] then
           local _,_,icap_type,_,icap_threat =
             headers['x-infection-found']:find("Type=(.-); Resolution=(.-); Threat=(.-);$")
 
-          if not icap_type or icap_type == 2 then
+          if icap_type == 2 then
```

Comment:
I don't see the way how it should work. Lua is not a JavaScript with shitty implicit conversions between strings and numbers. In this case, you compare a string `icap_type` with a number `2`. It is always false.